### PR TITLE
JBPM-8363: EJB Timer error in development mode for existing process during deploy

### PR DESCRIPTION
**Pull request by pefernan**
@jstastny-cz @mswiderski could you please take a look?

This is part of an ensemble please merge with:
https://github.com/kiegroup/jbpm/pull/1466
https://github.com/kiegroup/droolsjbpm-integration/pull/1783

### CODE REVIEW

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java [hunk 1]
```diff
@@ -496,6 +497,9 @@ public boolean isUpdateContainerAllowed(String id, KieContainerInstance kieConta
     public void updateContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
         // essentially it's a redeploy to make sure all components are up to date,
         // though update of kie base is done only once on kie server level and KieContainer is reused across all extensions
+        // we add this UPDATING_CONTAINER_PARAM
+        parameters.put(UPDATING_CONTAINER_PARAM, Boolean.TRUE);
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java]: Maybe use such parameter more for the dispose container so the condition would abortInstances or dispose. That way it’s more readable in my opinion

Comment (pefernan) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java]: @mswiderski you're right, maybe it is not clear enough.. Better now?

**Comment by MarianMacik** [comment]
Just so I understand it:

Before when a pure dispose was made, I had 2 options:
* Processes weren't aborted - default option
* Processes were aborted if `KieServerConstants.KIE_SERVER_PARAM_RESET_BEFORE_UPDATE` was explicitly set to true.

Now, when I purely dispose a container, `IS_DISPOSE_CONTAINER_PARAM` will be always true, so processes will be aborted regardless what is the value of `KieServerConstants.KIE_SERVER_PARAM_RESET_BEFORE_UPDATE`.

Is that right? The behaviour is changing a little bit.

**Comment by pefernan** [comment]
@MarianMacik 
That's right. The issue was that on dev mode when disposing a container (not updating) the active process instances weren't aborted. 
The point is that disposeContainer is called when disposing a container or when updating (dispose + create) so with this param we differenciate if this is a disposal or an update.

**Pull request merged**